### ISSUE TITLE
Bug introduced in 0.0.158

### ISSUE THREAD

**Issue by chasemcdo** [hwchase17/langchain#4129]
Updates in version 0.0.158 have introduced a bug that prevents this import from being successful, while it works in 0.0.157

```
Traceback (most recent call last):
  File "path", line 5, in <module>
    from langchain.chains import OpenAIModerationChain, SequentialChain, ConversationChain
  File "/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/langchain/__init__.py", line 6, in <module>
    from langchain.agents import MRKLChain, ReActChain, SelfAskWithSearchChain
  File "/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/langchain/agents/__init__.py", line 2, in <module>
    from langchain.agents.agent import (
  File "/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/langchain/agents/agent.py", line 15, in <module>
    from langchain.agents.tools import InvalidTool
  File "/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/langchain/agents/tools.py", line 8, in <module>
    from langchain.tools.base import BaseTool, Tool, tool
  File "/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/langchain/tools/__init__.py", line 32, in <module>
    from langchain.tools.vectorstore.tool import (
  File "/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/langchain/tools/vectorstore/tool.py", line 13, in <module>
    from langchain.chains import RetrievalQA, RetrievalQAWithSourcesChain
  File "/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/langchain/chains/__init__.py", line 19, in <module>
    from langchain.chains.loading import load_chain
  File "/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/langchain/chains/loading.py", line 24, in <module>
    from langchain.chains.sql_database.base import SQLDatabaseChain
  File "/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/langchain/chains/sql_database/base.py", line 15, in <module>
    from langchain.sql_database import SQLDatabase
  File "/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/langchain/sql_database.py", line 8, in <module>
    from sqlalchemy import (
ImportError: cannot import name 'CursorResult' from 'sqlalchemy' (/Users/chasemcdo/.pyenv/versions/3.11.1/lib/python3.11/site-packages/sqlalchemy/__init__.py)
```

**Comment by cancan101** [comment]
I think it is the change here: https://github.com/hwchase17/langchain/commit/fba6921b5093d7e7825d6eebc9234df92569b0ba#diff-177c01116eb534e9dfd4a4ee746ef865ccaadc8339ef1c300eb8e1e565155b26R8-R16 which requires sqla >= 1.4 whereas `pyproject.toml` specifies >=1.3

**Comment by lopez86** [comment]
I didn't notice that, I should be able to change it to import from sqlalchemy.engine instead of from sqlalchemy and bring the limit back down to 1.4

**Comment by fxrcode** [comment]
After installed SQLAlchemy 2+, the issue resolved, although CursorResult existed in 1.4

**Comment by chasemcdo** [comment]
Resolved by #4132
